uiowa/uiowa
# Sitenow_migrate updates

Had some local updates I wanted to get in place, and start an area where migrate platform updates can be added.

To start, was looking at creating basic link replace/report/removal functionality in one spot that can be added and/or overridden as needed.

- [x] Option to remove links on pages created before a certain date (used in pharmacy, at least one other request has mentioned this)
- [x] Option to replace links from a known mapping, if that mapping is available prior to migration
- [x] Option to replace internal links which may have been updated as part of the migration process
- [x] Update link fix option to account for multiple migrations
- [x] ~Fix the process order for standard sitenow migrations, at least, while allowing for incomplete migrations (eg only articles, only pages, people and articles but no pages, etc)~ Set up to run pages as a migration, as well as potentially articles and/or people prior. If a sitenow standard migration needed to be run without pages, but with link-replace functionality, minor code alterations can be done locally to accommodate.

# Updates included in this PR

`linkReplaceTrait` has been updated for tools available to future custom migrations. By default, when running a `d7_page` migration, it will attempt to update links on any page, article, or person which may have updated node ids for their destination.

Lots of code cleanup, including the removal of two of the three event listeners.





# How to test


`git checkout sitenow-migrate-updates`
`git pull`
`composer install`
Sync a sitenow standard D7 site in DevDesktop
`blt ds --site=default`
`drush @default.local uli /admin/config/development/configuration/config-split`
Enable the Sitenow Migrate split
`drush @default.local cim -y`
Go to https://default.local.drupal.uiowa.edu/admin/config/system/sitenow-migrate
Complete the form and save (help text is available for configuring to your DevDesktop instance)
`drush @default.local ms` <- view available migrations and their content counts. `d7_files` will probably have a weird number, possibly negative--that's okay; we're not concerned with it at the moment.
You can run `d7_article` and/or `d7_person` with `drush @default.local mim d7_article` and `drush @default.local mim d7_person`, respectively.
After running the above migrations (or not) `drush @default.local mim d7_page`.
If everything runs without error, congratulations!
Spot check some migrated pages, see if there's any weirdness or not. Try clicking on some links, and see if they bring you to the correct page.
